src/Message/PurchaseRequest.php

```diff
@@ -138,7 +1446,201 @@ public function getData()
+            || ($this->getCard() !== null && $this->getCard()->getShippingState() !== null)
+            || ($this->getCard() !== null && $this->getCard()->getShippingCountry() !== null)
+            || $this->getAddressMatch() !== null
+            || ($this->getCard() !== null && $this->getCard()->getBillingAddress1() !== null)
+            || ($this->getCard() !== null && $this->getCard()->getBillingAddress2() !== null)
+            || $this->getBillingAddress3() !== null
+            || ($this->getCard() !== null && $this->getCard()->getBillingCity() !== null)
+            || ($this->getCard() !== null && $this->getCard()->getBillingPostcode() !== null)
+            || ($this->getCard() !== null && $this->getCard()->getBillingState() !== null)
+            || ($this->getCard() !== null && $this->getCard()->getBillingCountry() !== null)
+            || $this->getChallengeWindowSize() !== null
+            || $this->getCustomerAdditionalInformation() !== null
+            || $this->get3DsRequestAuthenticationMethodData() !== null
+            || $this->get3DsRequestAuthenticationMethod() !== null
+            || $this->get3DsRequestAuthenticationTime() !== null
+            || $this->getCustomerAccountCreatedIndicator() !== null
+            || $this->getCustomerAccountCreatedDate() !== null
+            || $this->getCustomerAccountChangedIndicator() !== null
+            || $this->getCustomerAccountChangedDate() !== null
+            || $this->getCustomerPasswordChangedIndicator() !== null
+            || $this->getCustomerPasswordChangedDate() !== null
+            || $this->getCustomerPurchasesInLast6Months() !== null
+            || $this->getCustomerAccountCardProvisionsLast24Hours() !== null
+            || $this->getCustomerAccountTransactionsLast24Hours() !== null
+            || $this->getCustomerAccountTransactionsLastYear() !== null
+            || $this->getCustomerPaymentMethodCreatedIndicator() !== null
+            || $this->getCustomerPaymentMethodCreatedDate() !== null
+            || $this->getShippingAddressFirstUsedIndicator() !== null
+            || $this->getShippingAddressFirstUsedDate() !== null
+            || $this->getShippingNameCustomerNameMatch() !== null
+            || $this->getCustomerHasSuspiciousActivity() !== null
+            || $this->getDeliveryEmail() !== null
+            || $this->getDeliveryTimeframeIndicator() !== null
+            || $this->getGiftCardAmount() !== null
+            || $this->getGiftCardCount() !== null
+            || $this->getGiftCardCurrency() !== null
+            || $this->getPurchasingPreOrder() !== null
+            || $this->getPreOrderDate() !== null
+            || $this->getCustomerHasPurchasedProductBefore() !== null
+            || $this->getShippingAddressIndicator() !== null
+        ) {
+            $emv_3ds_parameters = array_filter([
+                'cardholderName'      => $this->getCardholder(),
```

Comment:
While there is a separate setter since the card object may not exist, if the card does exist (to supply address, etc.), then that could also potentially supply the cardholder name.